nixos/modules/services/misc/gitea.nix

```diff
@@ -454,6 +471,38 @@ in
       ];
     };
 
+    services.nginx = lib.mkIf cfg.configureNginx {
+      enable = true;
+      virtualHosts."${cfg.domain}" = {
+        root = "/var/empty";
+
+        locations."/" = {
+          proxyPass = (if cfg.enableUnixSocket then "http://unix:/run/gitea/gitea.sock" else "http://127.0.0.1:${toString cfg.httpPort}");
+          recommendedProxySettings = true;
+        };
+
+        locations."^~ /avatars/" = {
+          alias = "${cfg.stateDir}/data/avatars/";
+          tryFiles = "$uri =404";
+
+          extraConfig = ''
+            default_type "image";
+          '' + nginxCommonHeaders;
+        };
+
+        locations."^~ ${cfg.settings.server.STATIC_URL_PREFIX}/assets/" = {
+          alias = "${cfg.package.data}/public/";
+          tryFiles = "$uri =404";
```

Comment:
If possible, I always prefer to serve static files directly through nginx. In this case the file will be transferred a little faster.

Reply:
`/avatars/` (and `/repo-avatars/`, for that matter) isn't static, though.
It dynamically resizes the image based on the `?size` uri parameter.

And can be overwritten via `picture.AVATAR_UPLOAD_PATH` (`picture.REPOSITORY_AVATAR_UPLOAD_PATH`), which would break `"${cfg.stateDir}/data/avatars/"`.

Furthermore, those could be stored remotely using an entirely different storage driver using `AVATAR_STORAGE_TYPE` (`REPOSITORY_AVATAR_STORAGE_TYPE`).

I stand by what I said and do not understand why this nixos/gitea module needs to be made more complex than it needs to be.

Reply:
I can later try to test the operation with these parameters.

Reply:
If Gravatar is enabled, then `/avatar/<email-hash>?size=96` redirects (http/302) to `https://secure.gravatar.com/avatar/<email-hash>?d=identicon&s=96`

You are, however, correct, that the _local_ storage does not resize the images server-side (yet?[^1])
[^1]: Could change in the future, see https://github.com/go-gitea/gitea/issues/20918

Your avatar-serving breaks if
- Gravatar isn't disabled ([`DISABLE_GRAVATAR`](https://docs.gitea.io/en-us/config-cheat-sheet/#picture-picture)), which as of v1.18 moved to the database, meaning you can no longer `assert` it from the nixos config. **OR**
- `OFFLINE_MODE` is enabled, which also disables Gravatar as side effect. **OR**
- `AVATAR_STORAGE_TYPE`/`REPOSITORY_AVATAR_STORAGE_TYPE` is to something custom/not _local_ 

Plus, the docs for [`STATIC_URL_PREFIX`](https://docs.gitea.io/en-us/config-cheat-sheet/#server-server) explicitly state
> [...] Avatar images are dynamic resources and still served by Gitea. [...]

Reply:
If gravatar is used, the link will have an address like this - `https://www.gravatar.com/avatar/...` As a result, the rule `location ^~ /avatars/ {` will not work for them.

> Plus, the docs for [`STATIC_URL_PREFIX`](https://docs.gitea.io/en-us/config-cheat-sheet/#server-server) explicitly state

If avatar handling in gitea changes in the future, we can fix the nginx configuration.

Added small fix.

Reply:
oh lol sorry -- my bad, yes, you are right.
I somehow overlooked that additional `s` in `/avatars/` compared to `/avatar/` 😅 

I would still prefer this to be dropped altogether.
But I also don't want to annoy you anymore regarding this.
Just keep it however you imagined it originally and let the other reviewers decide.
Sorry for the noise (and the whole back and forth)

Reply:
Your comments helped improve PR. I didn't take into account possible changes to the defaults when I created the PR.